LoopKitTests/CarbStoreTests.swift

```diff
@@ -5,443 +5,446 @@
 //  Copyright © 2018 LoopKit Authors. All rights reserved.
 //
 
-import XCTest
-import HealthKit
-import CoreData
-@testable import LoopKit
+// TODO: Determine how to make these tests valid with new pull synchronization since
+// these tests tightly tie together both persistence and synchronization tests at once
```

Comment:
Yeah, the actions of storing and uploading have been split apart (yay!), but this means that we have a significantly different set of interactions with the store, and probably new tests are warranted.

Reply:
Done. Updated these tests and added many new tests for all of the new query anchors and store query code.